uuid.go

```diff
@@ -156,6 +158,53 @@ func (u UUID) String() string {
 	return string(buf)
 }
 
+// Format implements fmt.Formatter for UUID values.
+//
+// The behavior is as follows:
+// The 'x' and 'X' verbs output only the hex digits of the UUID, using a-f for 'x' and A-F for 'X'.
+// The 'v', 's' and 'q' verbs returns the canonical RFC-4122 string representation.
+// All other verbs not handled directly by the fmt package (like %p) return an empty string.
+func (u UUID) Format(f fmt.State, c rune) {
+	switch c {
+	case 'x', 'X':
+		s := hex.EncodeToString(u.Bytes())
+		if c == 'X' {
+			s = strings.Map(toCapitalHexDigits, s)
+		}
+		_, _ = io.WriteString(f, s)
+	case 'v', 's':
+		// TODO: dylan.bourque - 2019-02-18
+		// . should we implement %#v? if so, what is "a Go-syntax representation" of a UUID?
```

Comment:
I'd say let's do it. I wonder if we need to coerce the value in to a `[Size]byte` and then return the `%#v` of that.

Reply:
I'll give that a shot and see what it feels like. Sounds like the right path.

Reply:
The `%v` and `%+v` verbs still return the default RFC 4122 string.  Which format would be preferable for `fmt.Sprintf("%#v", u)`?
```
[Size]byte(u):
    "[16]uint8{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef, 0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}"
u.Bytes():
    "[]byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef, 0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}"
```

FWIW, I'm partial to the latter.

Reply:
`%#v` means Go syntax.

If `%#v` produced the string `"foo"` for type `Foo`, users should be able to type `var f Foo = foo` in their Go programs, and that code should compile.

Reply:
We'd want the `[16]uint8` format, because that's the real type:

```
const Size = 16

type UUID [Size]byte
```

Reply:
Under that rule, the output should be `uuid.UUID([]byte{.....})`, right?  But then it's "broken" if someone aliases the package name on import. :man_shrugging: #catch22

Reply:
Step 1: https://play.golang.org/p/ocJNNBjEiI2

Step 2: https://play.golang.org/p/s_dA-yquJZd

Since Step 2 compiles, that seems like the only real option to me.

Reply:
Implemented `%#v` as `[16]byte{....}`.

Also changed the default case to return `"%!verb(type=value)"` as recommended by the documentation in the `fmt` package (See "Format Errors" under https://golang.org/pkg/fmt/#hdr-Printing), i.e. `fmt.Sprintf("%d", u)` will return `%!d(uuid.UUID=....)`.

Reply:
> Also changed the default case to return `"%!verb(type=value)"` as recommended by the documentation in the fmt package

Very nice!